api/ask_astro/services/questions.py

```diff
@@ -27,6 +28,20 @@ async def _update_firestore_request(request: AskAstroRequest) -> None:
     )
 
 
+def _preprocess_request(request: AskAstroRequest) -> None:
+    if len(request.prompt) > 20000:
+        error_msg = "Question text is too long. Please try making a new thread and shortening your question."
+        request.response = error_msg
+        raise Exception(error_msg)
```

Comment:
I might suggestion adding a custom exception instead of using `Execption`

Reply:
+1